selfservice/strategy/passkey/passkey_login.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	if len(p.Login) > 0 || p.Method == s.SettingsStrategyID() {
+		// This method has only two submit buttons
+		p.Method = s.SettingsStrategyID()
+	} else {
+		return nil, flow.ErrStrategyNotResponsible
+	}
+
+	if err := flow.MethodEnabledAndAllowed(r.Context(), f.GetFlowName(), s.SettingsStrategyID(), p.Method, s.d); err != nil {
+		return nil, s.handleLoginError(r, f, err)
+	}
+
+	if err := flow.EnsureCSRF(s.d, r, f.Type, s.d.Config().DisableAPIFlowEnforcement(r.Context()), s.d.GenerateCSRFToken, p.CSRFToken); err != nil {
+		return nil, s.handleLoginError(r, f, err)
+	}
+
+	return s.loginPasswordless(w, r, f, &p)
+}
+
+func (s *Strategy) loginPasswordless(w http.ResponseWriter, r *http.Request, f *login.Flow, p *updateLoginFlowWithPasskeyMethod) (i *identity.Identity, err error) {
+	if err := login.CheckAAL(f, identity.AuthenticatorAssuranceLevel1); err != nil {
+		return nil, s.handleLoginError(r, f, err)
+	}
+
+	if err := flow.EnsureCSRF(s.d, r, f.Type, s.d.Config().DisableAPIFlowEnforcement(r.Context()), s.d.GenerateCSRFToken, p.CSRFToken); err != nil {
+		return nil, s.handleLoginError(r, f, err)
+	}
+
+	if len(p.Login) == 0 {
+		// Reset all nodes to not confuse users.
+		f.UI.Nodes = node.Nodes{}
+
+		err := s.populateLoginMethodForPasskeys(r, f)
+		if err != nil {
+			return nil, s.handleLoginError(r, f, err)
+		}
+		return nil, errors.WithStack(flow.ErrCompletedByStrategy)
+	}
+
+	return s.loginAuthenticate(w, r, f, p, identity.AuthenticatorAssuranceLevel1)
+}
+
+func (s *Strategy) loginAuthenticate(_ http.ResponseWriter, r *http.Request, f *login.Flow, p *updateLoginFlowWithPasskeyMethod, aal identity.AuthenticatorAssuranceLevel) (*identity.Identity, error) {
+	ctx := r.Context()
+
+	web, err := webauthn.New(s.d.Config().PasskeyConfig(ctx))
+	if err != nil {
+		return nil, s.handleLoginError(r, f, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to get webAuthn config.").WithDebug(err.Error())))
+	}
+
+	webAuthnResponse, err := protocol.ParseCredentialRequestResponseBody(strings.NewReader(p.Login))
+	if err != nil {
+		return nil, s.handleLoginError(r, f, errors.WithStack(herodot.ErrBadRequest.WithReasonf("Unable to parse WebAuthn response.").WithDebug(err.Error())))
+	}
+
+	var webAuthnSess webauthn.SessionData
+	if err := json.Unmarshal([]byte(gjson.GetBytes(f.InternalContext, flow.PrefixInternalContextKey(s.ID(), InternalContextKeySessionData)).Raw), &webAuthnSess); err != nil {
+		return nil, s.handleLoginError(r, f, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Expected WebAuthN in internal context to be an object but got: %s", err)))
+	}
+	webAuthnSess.UserID = nil
+
+	i, _, err := s.d.PrivilegedIdentityPool().FindByCredentialsIdentifier(r.Context(), s.ID(), string(webAuthnResponse.Response.UserHandle))
+	if err != nil {
+		time.Sleep(x.RandomDelay(s.d.Config().HasherArgon2(r.Context()).ExpectedDuration, s.d.Config().HasherArgon2(r.Context()).ExpectedDeviation))
```

Comment:
I don't think this is needed here. We're looking for the identity based on the user handle which is a randomly generated string. If this was an email, this could make sense here.

Reply:
makes sense, fixed :)